internal/storage/image_test.go

```diff
@@ -618,6 +620,40 @@ var _ = t.Describe("Image", func() {
 			Expect(res).To(BeNil())
 		})
 
+		It("should succeed to not list intermediate images", func() {
+			// Given
+			mockLoop := func() mockSequence {
+				return inOrder(
+					// buildImageCacheItem:
+					mockNewImage(storeMock, testSHA256, testSHA256),
+					// makeRepoDigests:
+					storeMock.EXPECT().ImageBigDataDigest(testSHA256, gomock.Any()).
+						Return(digest.Digest(""), nil),
+				)
+			}
+			inOrder(
+				storeMock.EXPECT().Images().Return(
+					[]cs.Image{
+						{ID: testSHA256, Names: []string{"a", "b", "c@sha256:" + testSHA256}},
+						{ID: testSHA256}},
+					nil),
+				// creates a layer with the parent of one of the images just returned
+				storeMock.EXPECT().Layers().Return([]cs.Layer{{Parent: testSHA256}}, nil),
+				mockParseStoreReference(storeMock, "@"+testSHA256),
```

Comment:
aahh sorry, missed this. Why not just have this in the mockLoop above?

Reply:
I think the mockLoop gets called many times, but I only need one layer

Reply:
I'm referring to mockParseStoreReference() which gets called above each mockLoop

Reply:
!!

Reply:
fixed !